e2e/e2etests/test_solana_withdraw_and_call_alt.go

```diff
@@ -0,0 +1,142 @@
+
+	// parse withdraw amount (in lamports), approve amount is 1 SOL
+	approvedAmount := new(big.Int).SetUint64(solana.LAMPORTS_PER_SOL)
+	require.Equal(
+		r,
+		-1,
+		withdrawAmount.Cmp(approvedAmount),
+		"Withdrawal amount must be less than the approved amount: %v",
+		approvedAmount,
+	)
+
+	// check balances before withdraw
+	connectedPda, err := solanacontract.ComputeConnectedPdaAddress(r.ConnectedProgram)
+	require.NoError(r, err)
+
+	connectedPdaInfoBefore, err := r.SolanaClient.GetAccountInfo(r.Ctx, connectedPda)
+	require.NoError(r, err)
+
+	// in case AddressLookupTable address is not provided (eg. for local testing) use prefunded random accounts to create AddressLookupTable
+	randomWallets := []solana.PublicKey{}
+	if initAddressLookupTable {
+		accounts := []solana.PublicKey{}
+
+		accounts = append(accounts, connectedPda)
+		accounts = append(accounts, r.ComputePdaAddress())
+		accounts = append(accounts, solana.SystemProgramID)
+		accounts = append(accounts, solana.SysVarInstructionsPubkey)
+		predefinedAccountsLen := len(accounts)
+		writableIndexes = []uint8{0} // only first one is mutable
+
+		addressLookupTableAddress, randomWallets = r.SetupTestAddressLookupTableWithRandomWallets(accounts)
+
+		// based on example accounts from above, all random wallets are writable
+		// since they will get some lamports from connected program example
+		for i := range randomWallets {
+			// #nosec G115 e2eTest - always in range
+			writableIndexes = append(
+				writableIndexes,
+				uint8(i+predefinedAccountsLen),
+			)
+		}
+	}
+
+	msg := solanacontract.ExecuteMsgAddressLookupTable{
+		AddressLookupTableAddress: [32]byte(addressLookupTableAddress),
+		WritableIndexes:           writableIndexes,
+		Data:                      []byte("hello"),
+	}
+
+	encoded, err := msg.Encode()
+	require.NoError(r, err)
+
+	randomWalletsBalanceBefore := []uint64{}
+	for _, acc := range randomWallets {
+		balanceBefore, err := r.SolanaClient.GetAccountInfo(r.Ctx, acc)
+		require.NoError(r, err)
+		randomWalletsBalanceBefore = append(randomWalletsBalanceBefore, balanceBefore.Value.Lamports)
+	}
+
+	// withdraw and call
+	tx := r.WithdrawAndCallSOLZRC20(
+		withdrawAmount,
+		approvedAmount,
+		encoded,
+		gatewayzevm.RevertOptions{
+			OnRevertGasLimit: big.NewInt(0),
+		},
+	)
+
+	// wait for the cctx to be mined
+	cctx := utils.WaitCctxMinedByInboundHash(r.Ctx, tx.Hash().Hex(), r.CctxClient, r.Logger, r.CctxTimeout)
+	utils.RequireCCTXStatus(r, cctx, crosschaintypes.CctxStatus_OutboundMined)
+
+	// get ERC20 SOL balance after withdraw
+	balanceAfter, err := r.SOLZRC20.BalanceOf(&bind.CallOpts{}, r.EVMAddress())
+	require.NoError(r, err)
+	r.Logger.Info("runner balance of SOL after withdraw: %d", balanceAfter)
+
+	// check if the balance is reduced correctly
+	amountReduced := new(big.Int).Sub(balanceBefore, balanceAfter)
+	require.True(r, amountReduced.Cmp(withdrawAmount) >= 0, "balance is not reduced correctly")
```

Comment:
Dont think we need to change in this PR , but wondering if we should make these checks a bit more strict , the two values should be similar